test/e2e/auth/node_authn.go

```diff
@@ -107,3 +104,30 @@ func createNodeAuthTestPod(f *framework.Framework) *v1.Pod {
 
 	return f.PodClient().CreateSync(pod)
 }
+
+// Execute a curl request from the pod to the url with the given headers.
+// Expect the result to either fail to connect to the host, or be a Unauthorized
+// or Forbidden response code.
+func expectUnauthorizedOrFailToConnect(f *framework.Framework, pod *v1.Pod, url string, headers ...string) {
+	headerArgs := "" // FIXME - don't log this
+	for _, header := range headers {
+		headerArgs = headerArgs + "--header " + header + " "
+	}
+
+	ns := f.Namespace.Name
+	result, err := framework.RunHostCmd(ns, pod.Name, fmt.Sprintf("curl -sIk -o /dev/null -w '%%{http_code}' %s %s", headerArgs, url))
+	if err != nil {
+		// Accept failure to connect as a success condition, since it indicates the port is
```

Comment:
Should this be handled by skipping the test?

Reply:
That would require breaking each port into a separate testcase. Doing so would make the logs more clear, but add a lot more tests (more latency). I'd prefer to leave it as is.